### PR TITLE
Exposes ov::Symbol and new methods of ov::Dimension

### PR DESCRIPTION
**Pull request by jane-intel**
### Details:
 - *Exposes ov::Symbol to py API*
 - *Exposes new ov::Dimension methods to py API*

### Tickets:
 - *CVS-136760*

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/graph/dimension.cpp [hunk 1]
```diff
@@ -121,6 +121,34 @@ void regclass_graph_Dimension(py::module m) {
...
+              :return: True if symbol was set, else False.
+              :rtype: bool
+            )");
+    dim.def("get_symbol",
+            &ov::Dimension::get_symbol,
+            R"(
+                Return this dimension's symbol as Symbol object.
+
+                :return: Value of the dimension.
+                :rtype: Symbol
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/dimension.cpp:139]:
```suggestion
                :rtype: openvino.Symbol
```

File: src/bindings/python/src/pyopenvino/graph/dimension.cpp [hunk 2]
```diff
@@ -121,6 +121,34 @@ void regclass_graph_Dimension(py::module m) {
...
+            )");
+
+    dim.def("set_symbol",
+            &ov::Dimension::set_symbol,
+            py::arg("symbol"),
+            R"(
+                Sets provided Symbol as this dimension's symbol.
+                :param symbol: The symbol to set to this dimension.
+                :type symbol: Symbol
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/dimension.cpp:148]:
```suggestion
                :type symbol: openvino.Symbol
```

File: src/bindings/python/src/pyopenvino/graph/symbol.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    symbol.def(py::init([]() {
+        return std::make_shared<ov::Symbol>();
+    }));
+
+    symbol.def(
+        "__eq__",
+        [](const std::shared_ptr<ov::Symbol>& a, const std::shared_ptr<ov::Symbol>& b) {
+            return ov::symbol::are_equal(a, b);
+        },
+        py::is_operator());
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/symbol.cpp:26]: How about other function like `ancestor_of`? Are there any reasons to not include them? https://github.com/openvinotoolkit/openvino/blob/c3c409ee133ffb26bf8fd5570ef50a7c004839a4/src/core/include/openvino/core/symbol.hpp#L15-L22

**Review by jiwaszki — APPROVED** [review]
Overall LGTM, small remarks.

File: src/bindings/python/src/pyopenvino/graph/symbol.cpp (hunk 3 above)
Comment (jane-intel) [reply src/bindings/python/src/pyopenvino/graph/symbol.cpp:26]: These are in the development mode, so we'd like to secure python API from frequent changes.
Additionally, user scenarios that we can think of don't require these functions.

File: src/bindings/python/src/pyopenvino/graph/dimension.cpp [hunk 4]
```diff
@@ -121,6 +121,34 @@ void regclass_graph_Dimension(py::module m) {
                 :return: Value of the dimension.
                 :rtype: int
             )");
+
+    /// Symbol-related methods: START
+    dim.def("has_symbol",
+            &ov::Dimension::has_symbol,
+            R"(
+              Check if Dimension has meaningful symbol.
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/dimension.cpp:129]:
```suggestion
              Check if Dimension has meaningful symbol.

```
Need to add space for proper parsing by doc builder

File: src/bindings/python/src/pyopenvino/graph/dimension.cpp [hunk 5]
```diff
@@ -121,6 +121,34 @@ void regclass_graph_Dimension(py::module m) {
...
+
+                :return: Value of the dimension.
+                :rtype: openvino.Symbol
+            )");
+
+    dim.def("set_symbol",
+            &ov::Dimension::set_symbol,
+            py::arg("symbol"),
+            R"(
+                Sets provided Symbol as this dimension's symbol.
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/dimension.cpp:146]:
```suggestion
                Sets provided Symbol as this dimension's symbol.

```

File: src/bindings/python/src/pyopenvino/graph/dimension.cpp [hunk 6]
```diff
@@ -121,6 +121,34 @@ void regclass_graph_Dimension(py::module m) {
...
+                :param symbol: The symbol to set to this dimension.
+                :type symbol: openvino.Symbol
+            )");
+    /// Symbol-related methods: END
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/dimension.cpp:152]: I suggest also to add property `symbol`
@jiwaszki what do you think?

Comment (jane-intel) [reply src/bindings/python/src/pyopenvino/graph/dimension.cpp:152]: Will it allow to get and set the value through it? If so, I would need an example of how to set it up.

Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/graph/dimension.cpp:152]: Let's not duplicate/simplify API for now. As now it is aligned, it's good enough.

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/graph/dimension.cpp:152]: yep,
please use [this ](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp#L124) as example

**Review by praasz — APPROVED** [review]

File: src/bindings/python/src/pyopenvino/graph/dimension.cpp (hunk 6 above)
Comment (jane-intel) [reply src/bindings/python/src/pyopenvino/graph/dimension.cpp:152]: got it. then no simplifications for now, will think about it in the future.

**Review by akuporos — APPROVED** [review]

**Review by p-durandin — APPROVED** [review]

**Pull request merged**